rust/kernel/sysctl.rs

```diff
@@ -110,17 +110,12 @@ unsafe extern "C" fn proc_handler<T: SysctlStorage>(
         return 0;
     }
 
-    let data = unsafe { UserSlicePtr::new(buffer, *len) };
+    let mut data = unsafe { core::slice::from_raw_parts_mut(buffer as *mut u8, *len) };
```

Comment:
Is the length checked to be correct by the caller of `proc_handler?

Reply:
https://github.com/Rust-for-Linux/linux/blob/rust/fs/proc/proc_sysctl.c#L570-L611

The length plus one is the byte size used to allocate the buffer. It is obtained
from the user iov and there’s a sanity check against ``KMALLOC_MAX_SIZE``.